lib/rpc/proto/utils.go

```diff
@@ -69,9 +71,51 @@ func EncodeError(err error) *Error {
 }
 
 func ErrorToMessage(err error) *Message {
-	return &Message{&Message_Error{EncodeError(err)}}
+	return &Message{Element: &Message_Error{EncodeError(err)}}
 }
 
 func DecodeError(err *Error) error {
 	return errors.New(err.Message)
 }
+
+func FormatPeerJoinRequest(req *PeerJoinRequest) string {
```

Comment:
Nit: Can't all these structs implement `Stringer` instead of having `Format*` methods?

Reply:
You're correct - I did not think about replacing the auto-generated `String` implementation with custom ones.